mathesar_ui/src/layouts/LayoutWithHeader.svelte

```diff
@@ -1,30 +1,67 @@
 <script lang="ts">
   import AppHeader from '@mathesar/components/AppHeader.svelte';
+  import LiveDemoBanner from '@mathesar/components/LiveDemoBanner.svelte';
 
   export let fitViewport = false;
+  export let restrictWidth = false;
+  export let cssVariables: Record<string, string> | undefined = undefined;
+
+  $: style = cssVariables
+    ? Object.entries(cssVariables)
+        .filter((val) => val[0].indexOf('--') === 0)
+        .map((entry) => `${entry[0]}: ${entry[1]}`)
+        .join(';')
+    : undefined;
 </script>
 
-<div class="app-layout" class:fit-viewport={fitViewport}>
+<div class="app-layout" class:fit-viewport={fitViewport} {style}>
+  <LiveDemoBanner />
   <div class="app-layout-header">
     <AppHeader />
   </div>
-  <main class="app-layout-content">
+  <slot name="secondary-header" />
+  <main class="app-layout-content" class:restrict-width={restrictWidth}>
     <slot />
   </main>
 </div>
 
-<style>
+<style lang="scss">
   .app-layout {
-    display: grid;
-    grid-template: auto auto / 1fr;
-  }
-  .app-layout.fit-viewport {
-    grid-template: auto 1fr / 1fr;
-    height: 100vh;
-  }
-  .app-layout-content {
-    position: relative;
+    height: 100%;
+    display: flex;
+    flex-direction: column;
+
+    .app-layout-header {
+      position: sticky;
```

Comment:
I notice the navigation header is now sticky. I have a slight preference for keeping it static because it allows more screen space on pages like the record page. I'm curious: was this a decision that you improvised in this PR, or did @ghislaineguerin specify a sticky header in some design document?

Reply:
This was mentioned by @ghislaineguerin in our figma design file, and we already had a position:sticky to the app header in the LayoutWithHeader2 component, but it didn't take effect because it did not have a position to stick to. I only added `top:0`.